### PR TITLE
Upgrade bokeh/bokehjs to 2.4

### PR DESCRIPTION
**Pull request by stefanv**
BokehJS now uses more JS features, so added the babel plugins for:

- optional chaining (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)
- nullish coalescing operator (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator).

Tweaked the models to be in line with the latest BokehJS (mainly usage
of `super.render()`).

<strike>Depends on https://github.com/bokeh/bokeh/issues/11616 to be resolved.</strike>

### CODE REVIEW

**Comment by stefanv** [comment]
Follow-up requests is failing with:

`E       selenium.common.exceptions.MoveTargetOutOfBoundsException: Message: (248, 1247) is out of bounds of viewport width (1920) and height (1115)`

Unsure how this is related to Bokeh, but seems to be happening consistently after restarting the tests.

File: static/js/components/BokehModels.js [hunk 1]
```diff
@@ -3,19 +3,18 @@
 import { input, label, div, span } from "bokehjs/core/dom";
 import { includes } from "bokehjs/core/util/array";
 import * as p from "bokehjs/core/properties";
-import { bk_inline } from "bokehjs/styles/mixins";
-import { bk_input_group } from "bokehjs/styles/widgets/inputs";
+import * as inputs from "bokehjs/styles/widgets/inputs.css";
```
Comment (acrellin) [inline static/js/components/BokehModels.js:6]:
```suggestion
import * as bokehInputsCss from "bokehjs/styles/widgets/inputs.css";
```
?

Comment (stefanv) [reply static/js/components/BokehModels.js:6]: This file is a modified version of code in Bokeh (checkbox widget) so I am trying to keep them the same with minimal changes to make it easier to diff.

Comment (acrellin) [reply static/js/components/BokehModels.js:6]: Okay, no problem then.

**Review by acrellin — APPROVED** [review]

**Pull request merged**